### PR TITLE
[Fix]  Fix label smoothing bug.

### PR DESCRIPTION
**Pull request by TeaPoly**
Fix bug when apply label smoothing. https://github.com/athena-team/athena/issues/322

### CODE REVIEW

**Comment by tjadamlee** [comment]
@Some-random please help check

File: athena/loss.py [hunk 1]
```diff
@@ -53,16 +53,20 @@ class Seq2SeqSparseCategoricalCrossentropy(tf.keras.losses.CategoricalCrossentro
...
         self.by_sequence = by_sequence
         self.num_classes = num_classes
         self.eos = num_classes + eos if eos < 0 else eos
+        self.smoothing_rate = label_smoothing
 
     def __call__(self, logits, samples, logit_length=None):
         labels = insert_eos_in_labels(samples["output"], self.eos, samples["output_length"])
         mask = tf.math.logical_not(tf.math.equal(labels, 0))
-        labels = tf.one_hot(indices=labels, depth=self.num_classes)
+        labels = apply_label_smoothing(
```
Comment (Some-random) [inline athena/loss.py]: It's better to check if smoothing rate is 0 first to avoid redundant calculation. I believe label smoothing is also implemented this way in Tensorflow API

**Comment by Some-random** [comment]
> @Some-random please help check

Sure

**Review by Some-random — APPROVED** [review]
